.github/tdd.py

```diff
@@ -0,0 +1,126 @@
+import subprocess
+import sys
+
+ORG_NAME = "pulp"
+REPO_NAME = "pulp_ansible"
+APP_NAME = "pulp_ansible"
+
+
+def get_current_branch():
+    pid = subprocess.run(
+        "git branch --show-current", shell=True, stdout=subprocess.PIPE, check=True
+    )
+    branch_name = pid.stdout.decode("utf-8")
+    branch_name = branch_name.strip()
+    return branch_name
+
+
+def get_changed_files(pr_branch, target_branch="master"):
+    if os.environ.get("GITHUB_PR_NUMBER"):
+        url = f"https://patch-diff.githubusercontent.com/raw/{ORG_NAME}/{REPO_NAME}"
+        url += f"/pull/{os.environ['GITHUB_PR_NUMBER']}.diff"
+        print(f"GET diff from {url}")
+        cmd = f"curl -s -o /tmp/pr.diff {url}"
+        pid = subprocess.run(cmd, shell=True)
+        assert pid.returncode == 0
+        with open("/tmp/pr.diff", "r") as f:
+            raw = f.read()
+        filenames = raw.split("\n")
+        filenames = [x for x in filenames if x.startswith("---") or x.startswith("+++")]
+        filenames = [x.split(None, 1)[1] for x in filenames]
+        filenames = [x for x in filenames if not x.startswith("/dev/null")]
+        filenames = [x.lstrip("a/") for x in filenames]
+        filenames = [x.lstrip("b/") for x in filenames]
+        filenames = sorted(set(filenames))
+    else:
+        # diffs the local checkout ...
+        cmd = f'git diff --name-only "{pr_branch}".."{target_branch}"'
+        pid = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, check=True)
+        filenames = pid.stdout.decode("utf-8")
+        filenames = filenames.split("\n")
+        filenames = [x.strip() for x in filenames if x.strip()]
+    return filenames
+
+
+def verify_test_files_changed(changed_files):
+
+    # places where appplication source goes
+    app_paths = [f"{APP_NAME}/"]
+
+    # places where test code goes
+    test_paths = [f"{APP_NAME}/tests"]
+
+    def is_app_path(fn):
+        for ap in app_paths:
+            if fn.startswith(ap):
+                return True
+        return False
+
+    def is_test_path(fn):
+        for tp in test_paths:
+            if fn.startswith(tp):
+                return True
+        return False
+
+    # exit early if no non-test changed in the api code
+    app_changed = False
+    for cf in changed_files:
+        if is_app_path(cf) and not is_test_path(cf):
+            app_changed = True
+    if not app_changed:
+        return
+
+    # look for any changes to file in the tests dirs
+    tests_found = False
+    for cf in changed_files:
+        for tp in test_paths:
+            if cf.startswith(tp):
+                tests_found = True
+                break
+        if tests_found:
+            break
+
+    if not tests_found:
+        raise Exception("Tests should be added or modified with -every- PR!!!")
+
+
+def main():
+    logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+
+    # these users should not be checked
+    skip_users = ["dependabot[bot]", "patchback[bot]"]
```

Comment:
Our automated CI (release process, plugin template updates) has [pulpbot](https://github.com/pulpbot) open PRs as well, think it should also be in this list.